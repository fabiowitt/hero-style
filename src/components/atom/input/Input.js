import React from 'react';
import classNames from 'classnames';
import css from './Input.module.scss';

const Input = (props) => {
  const {disabled, type} = props;
  const isCheckOrRadio = type==='checkbox' || type==='radio';

  const classes = classNames({
    [css['c-input']]: true,
    [css['c-input--disabled']]: disabled,
    [css[`c-input__${type}`]]: isCheckOrRadio,
    [css[`c-input__${type}--disabled`]]: disabled && isCheckOrRadio
  });

  return <input
    className={classes}
    {...props}
  />
};

Input.defaultProps = {
  type: 'text'
};

export default Input;
