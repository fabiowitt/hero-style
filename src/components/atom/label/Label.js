import React from 'react';
import PropTypes from 'prop-types';

const Label = ({ name, children }) => {
  return <label htmlFor={name}>{children}</label>;
};

Label.propTypes = {
  name: PropTypes.string,
  children: PropTypes.string.isRequired
};
Label.defaultProps = {
  name: null
};

export default Label;