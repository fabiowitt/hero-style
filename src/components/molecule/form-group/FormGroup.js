import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './FormGroup.module.scss';

const FormGroup = ({ success, error, children }) => {
  const classes = classNames({
    [css['c-form-group']]: true,
    [css['c-form-group--error']]: error,
    [css['c-form-group--success']]: success
  });

  return <div className={classes}>{children}</div>;
};

FormGroup.propTypes = {
  error: PropTypes.bool,
  success: PropTypes.bool,
  children: PropTypes.instanceOf(Array)
};
FormGroup.defaultProps = {
  error: false,
  success: false,
  children: null
};

export default FormGroup;
