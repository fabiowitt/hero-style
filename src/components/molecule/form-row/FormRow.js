import React from 'react';
import PropTypes from 'prop-types';

import css from './FormRow.module.scss';

const FormRow = ({ children }) => {
  return <div className={css['c-form-row']}>{children}</div>;
};

FormRow.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object
  ]).isRequired
};

export default FormRow;
