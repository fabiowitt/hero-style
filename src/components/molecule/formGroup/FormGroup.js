import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Label from '../../atom/label/Label';
import Input from '../../atom/input/Input';

import css from './FormGroup.module.scss';

const FormGroup = ({ id, name, type, placeholder, label, success, error, message }) => {
  const classes = classNames({
    [css['c-form-group']]: true,
    [css['c-form-group--error']]: error,
    [css['c-form-group--success']]: success
  });

  return <div className={classes}>
    <Label name={name}>{label}</Label>
    <Input id={id} name={name} type={type} placeholder={placeholder}></Input>
    {message ? <small>{message}</small> : null }
  </div>;
};

FormGroup.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  success: PropTypes.bool,
  error: PropTypes.bool,
  message: PropTypes.string
};
FormGroup.defaultProps = {
  id: null,
  name: null,
  type: 'text',
  placeholder: null,
  success: false,
  error: false,
  message: null
};

export default FormGroup;