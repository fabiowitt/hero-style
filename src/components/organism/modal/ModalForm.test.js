import React from 'react';
import { action } from '@storybook/addon-actions';
import { Grid } from 'react-flexbox-grid';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ModalForm from './ModalForm';
import FormRow from '../../molecule/form-row/FormRow';
import FormGroup from '../../molecule/form-group/FormGroup';
import Label from '../../atom/label/Label';
import Input from '../../atom/input/Input';

describe('ModalForm', () => {
  it('should match with the snapshot', () => {
    const isOpen = true;

    const wrapper = shallow(
      <div id="root">
        <ModalForm
          open={isOpen}
          onConfirm={() => console.log('confirm')}
          onCancel={() => console.log('onCancel')}
          contentLabel="With Actions EmptyModal"
          title="EmptyModal Title"
          description="Lorem ipsum dolor sit amet, consectetur adipisicing."
          btnCancelText="Cancelling"
          btnConfirmText="Confirming"
          onRequestClose={() => console.log('onRequestClose')}
        >
          <Grid fluid style={{ padding: 0 }}>
            <FormRow>
              <FormGroup md>
                <Label name="name">Nome</Label>
                <Input
                  id="name"
                  name="name"
                  type="text"
                  placeholder="Ex: Banheiro"
                  onChange={action('onChange')}
                  value=""
                />
              </FormGroup>
            </FormRow>
            <FormRow>
              <FormGroup md>
                <Label name="idRefurbish">Nome</Label>
                <Input
                  id="idRefurbish"
                  name="idRefurbish"
                  type="text"
                  placeholder="Ex: Banheiro"
                  onChange={action('idRefurbish')}
                  value=""
                />
              </FormGroup>
            </FormRow>

            <FormRow>
              <FormGroup md>
                <Label name="observation">Observações</Label>
                <Input
                  id="observation"
                  name="observation"
                  placeholder="Ativo"
                  multiple
                  onChange={action('change')}
                  value=""
                />
              </FormGroup>
            </FormRow>
          </Grid>
        </ModalForm>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
