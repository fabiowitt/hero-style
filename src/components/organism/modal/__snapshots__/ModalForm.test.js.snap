// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalForm should match with the snapshot 1`] = `
<div
  id="root"
>
  <ModalForm
    btnCancelText="Cancelling"
    btnConfirmText="Confirming"
    contentLabel="With Actions EmptyModal"
    description="Lorem ipsum dolor sit amet, consectetur adipisicing."
    left={false}
    onCancel={[Function]}
    onConfirm={[Function]}
    onRequestClose={[Function]}
    open={true}
    right={true}
    title="EmptyModal Title"
  >
    <Grid
      fluid={true}
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <FormRow>
        <FormGroup
          error={false}
          icon={false}
          md={true}
          success={false}
        >
          <Label
            name="name"
          >
            Nome
          </Label>
          <Input
            dark={null}
            defaultValue={null}
            disabled={null}
            id="name"
            name="name"
            onChange={[Function]}
            parentRef={null}
            placeholder="Ex: Banheiro"
            type="text"
            value=""
          />
        </FormGroup>
      </FormRow>
      <FormRow>
        <FormGroup
          error={false}
          icon={false}
          md={true}
          success={false}
        >
          <Label
            name="idRefurbish"
          >
            Nome
          </Label>
          <Input
            dark={null}
            defaultValue={null}
            disabled={null}
            id="idRefurbish"
            name="idRefurbish"
            onChange={[Function]}
            parentRef={null}
            placeholder="Ex: Banheiro"
            type="text"
            value=""
          />
        </FormGroup>
      </FormRow>
      <FormRow>
        <FormGroup
          error={false}
          icon={false}
          md={true}
          success={false}
        >
          <Label
            name="observation"
          >
            Observações
          </Label>
          <Input
            dark={null}
            defaultValue={null}
            disabled={null}
            id="observation"
            multiple={true}
            name="observation"
            onChange={[Function]}
            parentRef={null}
            placeholder="Ativo"
            type="text"
            value=""
          />
        </FormGroup>
      </FormRow>
    </Grid>
  </ModalForm>
</div>
`;
